.github/workflows/ci.yml

```diff
@@ -29,7 +29,13 @@ jobs:
 
   check:
     needs: sanity
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
```

Comment:
According to the [documentation](https://docs.gitlab.com/ee/ci/yaml/#rulesif) it's possible to target a branch.

Should we build on every platforms only when we merge into main or stage?

Reply:
Yes, you right! I will enable multiple platforms for `check`, `test` and `doctest` jobs only on `main` and `stage` branch :)